components/ledger/pkg/ledger/query/worker.go

```diff
@@ -424,11 +370,13 @@ func buildData(
 	return logsData, nil
 }
 
-func (w *Worker) QueueLog(log *core.LogHolder) {
+func (w *Worker) QueueLog(ctx context.Context, log *core.LogHolder) error {
 	select {
-	case <-w.ctx.Done():
+	case <-w.stoppedChan:
```

Comment:
Should we wait on the ctx here too ?

Reply:
Good question.
We need to NOT cancel the log ingestion if the request is cancelled, as, at this moment, the cache is already filled with new information.

Reply:
But we should be able to properly cancel the request and release any potential used resources.

Reply:
Good point about the cache !

Reply:
If we update the cache after the log ingestion, we will lock account until the log ingestion.
It would cancel the benefits of the CRQS no?